### PR TITLE
fix storybook build, restore E2E specs, try CircleCI

### PR DESCRIPTION
**Pull request by thedavidprice**
- fixes Storybook bug #3115
- fixed webpack-dev-server v4 breaking `.open` config change #3116 
- restores E2E specs (and enables node_module caching): Storybook, Logger, Saving Data, Auth (experimenting)
- [...otherChangesToBeAdded]

### CODE REVIEW

**Comment by thedavidprice** [comment]
boom

File: tasks/e2e/cypress/integration/01-tutorial/tutorial.spec.js [hunk 1]
```diff
@@ -26,6 +26,14 @@ const BASE_DIR = Cypress.env('RW_PATH')
 describe('The Redwood Tutorial - Golden path edition', () => {
   // TODO: https://redwoodjs.com/tutorial/saving-data
   // TODO: https://redwoodjs.com/tutorial/administration
+  after(() => {
+    cy.exec(
+      `cd ${BASE_DIR}; git add . && git commit -a --message=01-tutorial`,
+      {
+        failOnNonZeroExit: false,
```
Comment (peterp) [inline tasks/e2e/cypress/integration/01-tutorial/tutorial.spec.js:33]: We should probably make it fail if this doesn't work.

**Review by peterp — COMMENTED** [review]
This looks good, just a note that it doesn't fix: #3115 since the command it runs is `yarn rw sb --build` (which works), the command that fails is `yarn rw sb`

**Comment by thedavidprice** [comment]
In all my testing, I've never experienced different behavior between running start-storybook or build-storybook. So very curious you were experiencing that 🤔 I could not replicate myself using rwfw project:sync -- they were both failing.

Regardless, using rwfw on this branch I can confirm start-storybook is working correctly.

If you think it's necessary/helpful, it looks like we could use `start-storyboo --ci --smoke-test` as a step in our GitHub workflow, which will a) not open a browser window and b) exit after a successful start up. In this case I'd need to make the option available with our RW `storybook` command.

File: tasks/e2e/cypress/integration/01-tutorial/tutorial.spec.js (hunk 1 above)
Comment (thedavidprice) [reply tasks/e2e/cypress/integration/01-tutorial/tutorial.spec.js:33]: Good idea. I don't believe it's critical, but the logic makes more sense! Doing now.

**Comment by thedavidprice** [comment]
> Also, I wonder why caching works so well here, but not on the "build, lint & test" workers.

^^ In my experience, caching has always worked well for Ubuntu containers. Windows has been the troublemaker. Are you seeing something different?

**Comment by peterp** [comment]
I guess I was wondering why it was disabled... I figured that it gave us problems, but it seems to work like a legend here.

**Pull request merged**